packages/app/src/cli/services/context/identifiers-extensions.ts

```diff
@@ -30,21 +36,13 @@ export async function ensureExtensionsIds(
   }: AppWithExtensions,
 ) {
   let remoteExtensions = initialRemoteExtensions
-  const validIdentifiers = options.envIdentifiers.extensions ?? {}
+  const identifiers = options.envIdentifiers.extensions ?? {}
   const localExtensions = options.app.allExtensions.filter((ext) => ext.isUUIDStrategyExtension)
 
-  const uiExtensionsToMigrate = getUIExtensionsToMigrate(localExtensions, remoteExtensions, validIdentifiers)
-  const flowExtensionsToMigrate = getFlowExtensionsToMigrate(localExtensions, dashboardOnlyExtensions, validIdentifiers)
-  const marketingActivityExtensionsToMigrate = getMarketingActivtyExtensionsToMigrate(
-    localExtensions,
-    dashboardOnlyExtensions,
-    validIdentifiers,
-  )
-  const paymentsExtensionsToMigrate = getPaymentsExtensionsToMigrate(
-    localExtensions,
-    dashboardOnlyExtensions,
-    validIdentifiers,
-  )
+  const uiExtensionsToMigrate = getModulesToMigrate(localExtensions, remoteExtensions, identifiers, UIModulesMap)
+  const flowExtensionsToMigrate = getModulesToMigrate(localExtensions, remoteExtensions, identifiers, FlowModulesMap)
+  const paymentsToMigrate = getModulesToMigrate(localExtensions, remoteExtensions, identifiers, PaymentModulesMap)
+  const marketingToMigrate = getModulesToMigrate(localExtensions, remoteExtensions, identifiers, MarketingModulesMap)
```

Comment:
I think you need to pass `dashboardOnlyExtensions` to both of these, as the migration happens for dashboard only extensions which are not included in the `remoteExtensions`. Run into this issue myself when working on migrating subscription link extensions.

cc @isaacroldan